js/packages/openinference-instrumentation-langchain/src/typeUtils.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * Utility function that uses the type system to check if a switch statement is exhaustive.
+ * If the switch statement is not exhaustive, there will be a type error caught in typescript
+ *
+ * See https://stackoverflow.com/questions/39419170/how-do-i-check-that-a-switch-block-is-exhaustive-in-typescript for more details.
+ */
+export function assertUnreachable(_: never): never {
+  throw new Error("Unreachable");
+}
+
+/**
+ * A type-guard function for checking if a value is an object with string keys
+ */
+export function isObject(value: unknown): value is Record<string, unknown> {
+  return (
+    typeof value === "object" &&
+    value != null &&
+    !Array.isArray(value) &&
+    Object.keys(value).every((key) => typeof key === "string")
```

Comment:
The function name doesn't match the implementation. Best to name it accordingly `isObjectWithStringKeys`

p.s. why is this check needed for string keys?

Reply:
This was because lc has things typed as Record<string, any> for many of the properties on a run, so just knowing it was an object wasn't enough to make the types happy, I'll give it another pass though to see if i can make this less strict and still get away with it

Reply:
updated!